src/app/shared/entity/provider/aks.ts

```diff
@@ -26,3 +26,67 @@ export class AKSCloudSpec {
   clientSecret?: string;
   resourceGroup?: string;
 }
+
+export class AKSClusterSpec {
+  location: string;
+  kubernetesVersion: string;
+  nodeResourceGroup?: string;
+  enableRBAC?: boolean;
+  managedAAD?: boolean;
+  dnsPrefix?: string;
+  fqdnSubdomain?: string;
+  fqdn?: string;
+  privateFQDN?: string;
+  machineDeploymentSpec?: AKSMachineDeploymentCloudSpec;
+  networkProfile?: AKSNetworkProfile;
+}
+
+export interface AKSMachineDeploymentCloudSpec {
```

Comment:
Why is this an interface while `AKSClusterSpec` is a class? IMO we should `class` for all because all of them are being used as DTO.